SCClassLibrary/Common/Control/Server.sc

```diff
@@ -1151,6 +1151,35 @@ Server {
 		this.newAllocators;
 	}
 
+	quitSync { |condition|
+		condition ?? { condition = Condition.new };
+		condition.test = false;
+		this.quit;
+		this.prWaitForPidRelease({
+			condition.test = true;
+			condition.signal
+		}, {
+			// we may end up here
+			// if the server happened to become unresponsive recently
+			// in that case we try to quit again
+			this.quit;
+			this.prWaitForPidRelease({
+				condition.test = true;
+				condition.signal
+			}, {
+				// as a last resort...
+				"Server '%' failed to quit".format(this.name).postln;
+				this.pid !? {
+					"Forcing process to stop via system command.".postln;
+					thisProcess.platform.killProcessByID(this.pid)
+				};
+				condition.test = true;
+				condition.signal
+			});
+		}, 4); // timeout is longer than ServerStatusWatcher waiting to declare the server unresponsive
```

Comment:
This is actually: `this.options.pingsBeforeConsideredDead`. Maybe use that?

Reply:
Hm, I don't think so? `-pingsBeforeConsideredDead` is a counter and is used while server is running. I was referring to https://github.com/supercollider/supercollider/blob/e9922139f8dddf95cfba4e6315727b4b1978fca8/SCClassLibrary/Common/Control/ServerStatus.sc#L103

Reply:
You are right. Could you be so kind and make this an explicit return value in ServerStatus? Whe I wrote it, I simply copied it over. If someone changes this value in the code, the test would subtly fail.

Reply:
Do you mean something like
```supercollider
AppClock.sched(timeoutBeforeConsideredDeadOnExit, { 
```
?